internal/model/model.go

```diff
@@ -637,73 +638,115 @@ func (m *Model) ClusterConfig(deviceID protocol.DeviceID, cm protocol.ClusterCon
 		}
 	}
 
-	if m.cfg.Devices()[deviceID].Introducer {
```

Comment:
This stuff is just moved unchanged into a method because model.ClusterConfig() was getting large.